stdlib/REPL/src/LineEdit.jl

```diff
@@ -432,12 +463,29 @@ prompt_string(p::Prompt) = prompt_string(p.prompt)
 prompt_string(s::AbstractString) = s
 prompt_string(f::Function) = Base.invokelatest(f)
 
+function maybe_show_hint(s::PromptState)
+    isa(s.hint, String) || return nothing
+    # The hint being "" then nothing is used to first clear a previous hint, then skip printing the hint
+    # the clear line cannot be printed each time because it breaks column movement
+    if isempty(s.hint)
+        print(terminal(s), "\e[0K") # clear remainder of line which had a hint
```

Comment:
I am experiencing frequent REPL corruption now where I lose part of the line or it prints wrong. For example, if you type
```
julia> using Cthu, REPL
```
then mouse back and change that to
```
julia> using Cthulhu, REPL
```
when you start typing, it will garble the rest of the line. Then when you type the final `u`, the line gets fully deleted. Once you type another character, then it gets restored correctly.

Reply:
I have a fix ready. Basically it limits to only doing hints for the end of the line. Inserting hints at earlier positions would take a lot of work in the printing logic I think so best to just limit it for now. 

Maybe it's possible to implement earlier hints before 1.11 is out.

Also I'm considering not hinting on single characters, given `x` shows `xor` as a hint which is a bit awkward.

Reply:
That seems useful. Could you also make the hint print with light colors?

Reply:
It already prints in `:light_black` which seems like the right choice?

Reply:
Oh, I see, I am using `TERM=screen-256color` and I don't have the terminal file for that (only for `screen-256color-s` with an added status line feature), so Julia is not printing any colors for me